### DISCUSSION TITLE
Provide ready-to-use implementations for oauth2-rs / openidconnect-rs

### DISCUSSION THREAD

**Discussion by yu-re-ka** [maxcountryman/axum-login#173] [Ideas]
I saw the example using oauth2-rs, and thought, this is exactly what I need: But now I need to copy all of this example code together, when this is really a quite generic implementation that could be re-used in many settings, given some parameters to set up the oauth2 client.
So maybe it would be possible to provide this oauth2 backend inside the axum-login crate behind a feature flag.

**Comment by maxcountryman** [comment]
This becomes more nuanced when we observe that AuthnBackend and optionally AuthzBackend are specific implementations. What this means is that once an application wants to do something like add another OAuth provider or support passworded authentication these implementations also need to change. Maintaining that flexibility while also offering concrete implementations would be tricky and I think it's likely it would fit better into a crate built on top of this one.
